docs/articles/documentation/extending-testcafe/custom-reporter/index.md

```diff
@@ -0,0 +1,232 @@
+* [Implementing Reporter](#implementing-reporter)
+* [Building Reporter](#building-reporter)
+* [Testing Reporter](#testing-reporter)
+* [Previewing Report](#previewing-report)
+* [Using Reporter Development Version](#using-reporter-development-version)
+* [Publishing Reporter to NPM](#publishing-reporter-to-npm)
+* [Reference](#reference)
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir my-reporter
+cd my-reporter
+```
+
+**Note:** It is recommended to name the directory as you would like to name the reporter project. When you run the generator, it will automatically suggest the reporter name that matches the reporter directory name. As for the reporter package name, it will consist of two parts - the `{testcafe-reporter-}` prefix and the name of the reporter itself:
+
+```text
+{testcafe-reporter-my-reporter}
+```
+
+Then run the reporter generator to create a new project:
+
+```bash
+yo testcafe-reporter
+```
+
+The generator will provide additional settings to customize your reporter.
+Then Yeoman will automatically scaffold out your reporter, install the required dependencies, and pull in several useful Gulp tasks for your workflow.
+
+## Implementing Reporter
+
+Once the reporter has been scaffolded out, go to the reporter directory and open the `src/index.js` file.
+
+You have to implement four reporter methods.
+
+```js
+reportTaskStart (/* startTime, userAgents, testCount */) {
+    throw new Error('Not implemented');
+},
+
+reportFixtureStart (/* name, path */) {
+    throw new Error('Not implemented');
+},
+
+reportTestDone (/* name, errs, durationMs, unstable, screenshotPath */) {
+    throw new Error('Not implemented');
+},
+
+reportTaskDone (/* endTime, passed */) {
+    throw new Error('Not implemented');
+}
+```
+
+These methods should output the desired information at certain moments during test run.
+
+* **reportTaskStart** - fires when a test task is started, that is at the beginning of testing.
+
+* **reportFixtureStart** - fires each time a fixture is started.
+
+* **reportTestDone** - fires each time a test is finished.
+
+* **reportTaskDone** - fires when the entire task is finished, that is at the end of testing.
+
+All the required data is provided to these methods through their parameters.
+
+To output information, use [helper methods](helper-methods.md) and libraries.
+TestCafe will mixin the helper methods to the reporter, so that you can access the helper methods by using `this`.
+
+In the `src/index.js` file, you can also enable or disable coloring a reporter output by using the `noColors` property. To color the output, use the [chalk](helper-methods.md#chalk) helper method.
```

Comment:
you can say
*enable or disable reporter output coloring*
or
*enable or disable coloring of the reporter output*